File: implement-strstr/implement-strstr.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function(haystack, needle) {
    if (needle.length == 0) return 0
    
    let window = needle.length
    
    for (let i = 0; i<haystack.length; i++){
        if (needle == haystack.slice(i,window+i)){
            return i
        }
    }
    return -1
};

File: implement-strstr/implement-strstr.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        if len(needle) == 0: return 0
        
        window = len(needle)
        
        for i, x in enumerate(haystack):
            if needle == haystack[i:window+i]: 
                return i        
        return -1